pkg/license/reporter.go

```diff
@@ -0,0 +1,73 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package license
+
+import (
+	"time"
+
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	// refreshPeriod defines how often the licensing information should be refreshed
+	refreshPeriod = 2 * time.Minute
+)
+
+var (
+	log = logf.Log.WithName("resource")
+)
+
+// ResourceReporter aggregates resources of all Elastic components managed by the operator
+// and reports them in a config map in the form of licensing information
+type ResourceReporter struct {
+	aggregator        Aggregator
+	licensingResolver LicensingResolver
+}
+
+// NewResourceReporter returns a new ResourceReporter
+func NewResourceReporter(client client.Client) ResourceReporter {
+	c := k8s.WrapClient(client)
+	return ResourceReporter{
+		aggregator: Aggregator{
+			client: c,
+		},
+		licensingResolver: LicensingResolver{
+			client: c,
+		},
+	}
+}
+
+// Start starts to report the licensing information repeatedly at regular intervals
+func (r ResourceReporter) Start(operatorNs string) {
+	ticker := time.NewTicker(refreshPeriod)
```

Comment:
would be nice to call `r.Report()` once as soon as possible so we don't wait the first 2 minutes if we just want to test it out. Cache should be sync already once this is called.